server/src/query/stream_schema_provider.rs

```diff
@@ -236,57 +236,55 @@ fn partitioned_files(
         count += num_rows;
     }
 
-    let mut statistics = vec![];
-
-    for field in table_schema.fields() {
-        let Some(stats) = column_statistics
-            .get(field.name())
-            .and_then(|stats| stats.as_ref())
-        else {
-            statistics.push(datafusion::common::ColumnStatistics::default());
-            break;
-        };
+    let statistics = table_schema
+        .fields()
+        .iter()
+        .map(|field| {
```

Comment:
The body of this closure is big enough to have a name. Can we have an independent function?

Reply:
I refactored it. PTAL